packages/editor/src/plugins/text/components/suggestions.tsx

```diff
@@ -1,65 +1,116 @@
+  }
+
+  const handleItemClick = (
+    event: React.MouseEvent<HTMLButtonElement>,
+    pluginType: EditorPluginType | string
+  ) => {
+    event.preventDefault()
+    insertSelectedPlugin(pluginType)
+  }
+
+  const handleItemFocus = (index: number) => {
+    setCurrentlyFocusedItem(index)
+  }
+
+  const handleItemBlur = () => {
+    setCurrentlyFocusedItem(-1)
+  }
 
-  if (options.length === 0) {
-    return <div>{editorStrings.plugins.text.noItemsFound}</div>
+  const renderSuggestions = (
+    options: SuggestionOption[],
+    offset: number = 0
+  ) => {
+    return options.map((item, index) => {
+      const currentIndex = index + offset
+      return (
+        <SuggestionItem
+          key={currentIndex}
+          option={item}
+          selected={currentIndex === currentlyFocusedItem}
+          ref={(el) => (itemRefs.current[currentIndex] = el)}
+          onClick={(event) => handleItemClick(event, item.pluginType)}
+          onFocus={() => handleItemFocus(currentIndex)}
+          onBlur={handleItemBlur}
+          onMouseEnter={() => handleItemFocus(currentIndex)}
+          onMouseLeave={() => {
+            handleItemBlur()
+            itemRefs.current[currentIndex]?.blur()
+          }}
+          tooltipPosition={getTooltipPosition(index)}
+        />
+      )
+    })
   }
 
   return (
-    <div ref={suggestionsRef} className="max-h-[387px] overflow-auto">
-      {options.map(({ pluginType, title, description, icon }, index) => {
-        return (
-          <div
-            key={index}
-            data-qa={`plugin-suggestion-${pluginType}`}
-            data-active={index === selected}
-            onMouseDown={(event: React.MouseEvent) => {
-              event.preventDefault()
-              onMouseDown(pluginType)
-            }}
-            onMouseMove={() => {
-              onMouseMove(index)
-            }}
-            className={cn(`
-              group/suggestion flex cursor-pointer items-center px-5 py-2.5
-              hover:bg-editor-primary-50 data-[active=true]:bg-editor-primary-50
-            `)}
-          >
-            <div
-              className={cn(`
-               mr-3 flex-[0_0_95px] rounded-md border border-transparent group-hover/suggestion:border-gray-300
-               group-data-[active=true]/suggestion:border-gray-300 [&>svg]:rounded-md
-              `)}
-            >
-              {icon ?? <IconFallback />}
-            </div>
-            <div>
-              <h5 className="text-base font-bold">{title}</h5>
-              <p className="whitespace-pre-wrap text-base">{description}</p>
-            </div>
-          </div>
-        )
-      })}
+    <div className="mt-2">
+      <div className="sticky top-0 z-10 bg-white pb-3 pl-6 pt-7 shadow-stickysearch">
+        <EditorInput
+          ref={searchInputRef}
+          autoFocus
+          placeholder="Search..."
+          value={searchString}
+          onChange={(e) => setSearchString(e.target.value)}
```

Comment:
Minor: On line 74 you used `event`, here you're using `e`. Sticking to one, one per file at least, would be nice.